src/Datadog.Trace.ClrProfiler.Managed/ScopeFactory.cs

```diff
@@ -38,11 +38,13 @@ public static Scope CreateOutboundHttpScope(Tracer tracer, string httpMethod, Ur
             try
             {
                 Span parent = tracer.ActiveScope?.Span;
+                string httpUrl = requestUri != null ? UriHelpers.CleanUri(requestUri, removeScheme: false, tryRemoveIds: false) : null;
```

Comment:
I think the code is much cleaner with this local, but I'm a bit concerned about calling ` UriHelpers.CleanUri()` every time even though it's only needed in some cases (it's the last comparison in the `if` below). How about getting the tags first so we can check them for null?

```csharp
string methodTag = parent.GetTag(Tags.HttpMethod);
string urlTag = parent.GetTag(Tags.HttpUrl);

if (parent != null &&
    parent.Type == SpanTypes.Http &&
    parent.GetTag(Tags.InstrumentationName) != null &&
    methodTag != null &&
    urlTag != null &&
    string.Equals(httpMethod, methodTag , StringComparison.OrdinalIgnoreCase) &&
    string.Equals(UriHelpers.CleanUri(...), urlTag, StringComparison.OrdinalIgnoreCase))
{
```

This way the `string.Equals()` and `UriHelpers.CleanUri()` calls only happen if the tags are not null. Maybe I'm overthinking this?

Reply:
You're right, in that iteration the `UriHelpers.CleanUri()` would be expensive and oftentimes unnecessary. I've removed the HTTP tag checks because we can still uniquely make this decision without that information

Reply:
Resolving this thread as it's no longer applicable